### ISSUE TITLE
Make a distinction between "Container group ports" and "Container ports" in azurerm_container_group

### ISSUE THREAD

**Issue by w0ut0** [hashicorp/terraform-provider-azurerm#6409]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I want to reopen issue [1662](https://github.com/terraform-providers/terraform-provider-azurerm/issues/1662), as I feel it has been closed prematurely without actually solving the issue.

The solution that closed the issue, did not leave room for the following scenario (with 2 containers):

- Open port 443 in an Nginx sidecar container
- Open port 80 in the webserver container
- Expose **only** port 443 on the public IP of the container group.

This solution is possible using Microsoft's ARM and YAML templates (see _properties.containers.0.properties.ports_ vs _properties.ipAddress.ports_ [here](https://docs.microsoft.com/en-us/azure/container-instances/container-instances-reference-yaml#schema)

### New or Affected Resource(s)

* azurerm_container_group

### Potential Terraform Configuration

```hcl
resource "azurerm_resource_group" "webapp" {
    name     = "webapp-rg"
    location = "west europe"
}

resource "azurerm_container_group" "webapp" {
    name                = "webapp-aci"
    location            = "${azurerm_resource_group.vsts_agent.location}"
    resource_group_name = "${azurerm_resource_group.vsts_agent.name}"
    os_type             = "linux"
    ip_address_type     = "public"
    ports {
        port     = 443
        protocol = "TCP"
    }

    container {
        name             = "nginx"
        image            = "nginx"
        cpu              = "1"
        memory           = "2"
        ports {
            port     = 443
            protocol = "TCP"
        }
    }
    container {
        name             = "webapp"
        image            = "webapp"
        cpu              = "1"
        memory           = "2"
        ports {
            port     = 80
            protocol = "TCP"
        }
    }
}
```

### References

* https://github.com/terraform-providers/terraform-provider-azurerm/issues/1662
* https://docs.microsoft.com/en-us/azure/container-instances/container-instances-reference-yaml#schema
* https://docs.microsoft.com/en-us/azure/templates/microsoft.containerinstance/containergroups#template-format

**Comment by amasover** [comment]
Hi @w0ut0 there is an issue open already #4413 to continue the work from #1662.

I also have a PR open #5091. My PR matches your suggested terraform configuration exactly.

The main problem with this change as I understand, is that it breaks users whose Terraform configuration previously only specified ports on the container level. After merging my PR, they would have to add a matching `ports` block on the container group level.

I'm hoping that I can get feedback from the maintainers on this. For now, because I myself have the need to deploy a container group w/ sidecar similar to what you've described, I build off my own branch like so: run `make build` and copy the binary to `~/.terraform.d/plugins/terraform-provider-azurerm_v9.9.9`. Then in the module, set required_providers block:

```
terraform {
  required_providers {
    azurerm = "9.9.9"
  }
}

**Comment by amasover** [comment]
This can probably be closed now that #10491 is merged.

**Comment by favoretti** [comment]
Since this issue seems to have been addressed in the latest versions of the provider - I'm going to close it. Please open a new updated bug report if this is still relevant. Thank you.
